src/bin/add/args.rs

```diff
@@ -106,6 +109,22 @@ impl Args {
 
         Ok(vec![dependency])
     }
+
+    fn get_upgrade_prefix(&self) -> &'static str {
```

Comment:
I also think it has something to do with the fact that the `map` function takes ownership of `self` which seems like a bad thing.